commit 4a9d2de43d7539b93263b7d641d54b9232ebab07
Author: Colin Ian King <colin.king@canonical.com>
Date:   Tue Oct 3 13:12:54 2017 +0100

    UBUNTU: SAUCE: LSM stacking: check for invalid zero sized writes
    
    BugLink: http://bugs.launchpad.net/bugs/1720779
    
    Writing zero bytes to /proc/$pid/task/$pid/attr/context via
    security_setprocattr cause an oops in memcpy_erms. Fix this by
    checking for zero size and returning -EINVAL for this invalid
    write size.
    
    Detected by running stress-ng --procfs 0
    
    Signed-off-by: Colin Ian King <colin.king@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/security/security.c b/security/security.c
index 29d43dae9a8a..4680ebd4dc73 100644
--- a/security/security.c
+++ b/security/security.c
@@ -1997,6 +1997,9 @@ int security_setprocattr(const char *lsm, const char *name, void *value,
 	char *temp;
 	char *cp;
 
+	if (!size)
+		return -EINVAL;
+
 	/*
 	 * If lsm is NULL look at all the modules to find one
 	 * that processes name. If lsm is not NULL only look at
